src/main/scala/org/zalando/zhewbacca/SecurityRule.scala

```diff
@@ -46,27 +46,35 @@ abstract case class ValidateTokenRule(
   */
 case class ExplicitlyAllowedRule(method: String, pathRegex: String) extends StrictRule(method, pathRegex) {
 
-  override def execute(nextFilter: (RequestHeader) => Future[Result], requestHeader: RequestHeader)(implicit ec: ExecutionContext): Future[Result] =
+  override def execute(nextFilter: RequestHeader => Future[Result], requestHeader: RequestHeader)(implicit ec: ExecutionContext): Future[Result] =
     nextFilter(requestHeader)
 
 }
 
 /**
   * Useful for explicitly denied HTTP methods or URIs.
   */
-case class ExplicitlyDeniedRule(method: String, pathRegex: String) extends StrictRule(method, pathRegex) {
-
-  override def execute(nextFilter: (RequestHeader) => Future[Result], requestHeader: RequestHeader)(implicit ec: ExecutionContext): Future[Result] =
-    Future.successful(Results.Forbidden)
-}
+case class ExplicitlyDeniedRule(authProvider: AuthProvider, method: String, pathRegex: String) extends StrictRule(method, pathRegex) with DenySecurityRule
 
 /**
   * Default rule for `SecurityFilter`.
   */
-class DenyAllRule extends SecurityRule {
-
+case class DenyAllRule(authProvider: AuthProvider) extends DenySecurityRule {
   override def isApplicableTo(requestHeader: RequestHeader): Boolean = true
+}
+
+trait DenySecurityRule extends SecurityRule {
+  def authProvider: AuthProvider
+
+  private[this] val log = Logger(this.getClass)
 
   override def execute(nextFilter: (RequestHeader) => Future[Result], requestHeader: RequestHeader)(implicit ec: ExecutionContext): Future[Result] =
-    Future.successful(Results.Forbidden)
+    RequestValidator.validate(Scope.Empty, requestHeader, authProvider).flatMap[Result] {
```

Comment:
I'd suggest to always return 403 in case of 'deny all' because the server won't authorize the client with any amount of subsequent attempts.
```
Error 403: "The server understood the request, but is refusing to fulfill it. Authorization will not help and the request SHOULD NOT be repeated."
```

Reply:
This is not of a huge importance, but the "spirit" of these error codes is like following:
401 - Hey, your credentials are invalid for some reason, get valid one and come back
403 - Hey, your valid credentials are not allowing you to access this, only come back after becoming more powerful

But in both cases the client is doing something wrong and might fix the situation in one way or another. In this specific case the client can to nothing about this which suggests the server should say something about the _resource_, not about the request